lib/THC/THCTensorMathBlas.cu

```diff
@@ -233,6 +235,45 @@ void THCudaTensor_addr(THCState *state, THCudaTensor *r_, float beta, THCudaTens
   }
 }
 
+void THCudaTensor_addbmm(THCState *state, THCudaTensor *result, float beta, THCudaTensor *t,
+    float alpha, THCudaTensor *batch1, THCudaTensor *batch2) {
+  THAssert(THCudaTensor_checkGPU(state, 4, result, t, batch1, batch2));
+  THArgCheck(THCudaTensor_nDimension(state, t) == 2, 4, "expected 2D tensor");
+  THArgCheck(THCudaTensor_nDimension(state, batch1) == 3, 6, "expected 3D tensor");
+  THArgCheck(THCudaTensor_nDimension(state, batch2) == 3, 7, "expected 3D tensor");
+
+  int batchnum = THCudaTensor_size(state, batch1, 0);
+  int m1d1 = THCudaTensor_size(state, batch1, 1);
+  int innerdim = THCudaTensor_size(state, batch1, 2);
+  int m2d2 = THCudaTensor_size(state, batch2, 2);
+
+  THArgCheck(batchnum == THCudaTensor_size(state, batch2, 0), 7,
+      "equal number of batches expected");
+  THArgCheck(m1d1 == THCudaTensor_size(state, t, 0), 6,
+      "wrong matrix size");
+  THArgCheck(m2d2 == THCudaTensor_size(state, t, 1), 7,
+      "wrong matrix size");
+  THArgCheck(innerdim == THCudaTensor_size(state, batch2, 1), 6,
+      "wrong matrix size");
+
+  if (t != result) {
+    THCudaTensor_resizeAs(state, result, t);
+    THCudaTensor_copy(state, result, t);
+  }
+
+  THCudaTensor_mul(state, result, result, beta);
```

Comment:
I think multiplying by 1 a bunch is faster than a random float, but I'll change this in case I'm doing premature optimization.